examples/numpy/air-quality-analysis/analysis_notebook.ipynb

```diff
@@ -0,0 +1,1135 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 1 - import the things you need\n",
+    "import logging\n",
+    "import sys\n",
+    "from functools import partial\n",
+    "import typing\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from numpy.random import default_rng\n",
+    "from scipy import stats\n",
+    "\n",
+    "from hamilton import ad_hoc_utils, driver, base\n",
+    "\n",
+    "logging.basicConfig(stream=sys.stdout)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 2 - import modules to create part of the DAG from\n",
+    "# We use the autoreload extension that comes with ipython to automatically reload modules when\n",
+    "# the code in them changes.\n",
+    "\n",
+    "# import the jupyter extension\n",
+    "%load_ext autoreload\n",
+    "# set it to only reload the modules imported\n",
+    "%autoreload 1\n",
+    "# import the function modules you want to reload when they change.\n",
+    "# i.e. these should be your modules you write your functions in. As you change them,\n",
+    "# they will be reimported without you having to do anything.\n",
+    "# %aimport analysis_flow "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cell 3 - Define your new Hamilton functions & curate them into a TemporaryFunctionModule object.\n",
+    "# This enables you to add functions to your DAG without creating a proper module.\n",
+    "# This is ONLY INTENDED FOR QUICK DEVELOPMENT. For moving to production move these to an actual module.\n",
+    "\n",
+    "\n",
+    "# We'll take out all the private functions into their own separate module\n",
```

Comment:
In Hamilton the convention is functions with an underscore are "private". So you don't want to stick them in a module :)

So these are fine as they are -- just remove adding them to the module (and thus the driver). Things should just work.

Reply:
Ah okay, I misunderstood the idea behind the temporary module technique, I assumed it was about logical separation. Have removed it and it works as before. 

I have also changed the name of the notebook to `hamilton_notebook.ipynb` so as to keep it consistent across all examples.